src/seeded_rng.rs

```diff
@@ -129,6 +129,14 @@ impl Rng for SeededRng {
             Some(&arg[index])
         }
     }
+
+    fn shuffle<T>(&mut self, values: &mut [T]) {
+        let mut i = values.len();
+        while i >= 2 {
+            i -= 1;
+            values.swap(i, self.gen_range(0, (i + 1) as u32) as usize);
+        }
+    }
```

Comment:
We could, but I'm not sure if it's more readable. The form I'm proposing is directly based on the original implementation: https://doc.rust-lang.org/rand/src/rand/lib.rs.html#554-563

As for doing `as u32) as usize)` - it's to guarantee that the same numbers will be generated on 32-bit and on 64-bit machines. When `usize`s are generated directly, the results are different and are causing some non-determinism in the tests across architectures.